src/main/kotlin/com/autonomousapps/tasks/SynthesizeProjectViewTask.kt

```diff
@@ -190,9 +190,11 @@ abstract class SynthesizeProjectViewTask @Inject constructor(
         .map { it.build() }
         .toSet()
 
-      val projectCoordinates = ProjectCoordinates(parameters.projectPath.get())
+      val projectCoordinates = ProjectCoordinates(parameters.projectPath.get(),
+        GradleVariantIdentification(emptySet(), emptyMap()))
+      val ignoreSelfDependencies = parameters.buildType.isPresent // ignore on Android
```

Comment:
clever

Reply:
I am actually not 100% sure why it needs to be like this on Android. But there were tests showing cases that needed this.

In "plain JVM" projects this cannot be ignored to support all cases that are shown in the tests this PR adds. So I think this distinction here is fine for now.